skimage/filters/tests/test_frangi.py

```diff
@@ -0,0 +1,30 @@
+import numpy as np
+from numpy.testing import assert_equal, assert_almost_equal
+from skimage.filters._frangi import frangi_filter, hessian_filter
+
+
+def test_null_matrix():
+    a = np.zeros((3, 3))
+    assert_almost_equal(frangi_filter(a), np.ones((3, 3))/450.5)
+    assert_equal(hessian_filter(a), np.ones((3, 3)))
+
+
+def test_energy_decrease():
+    a = np.zeros((3, 3))
+    a[1, 1] = 1.
+    frangi_a = frangi_filter(a)
+    hessian_a = hessian_filter(a)
+    assert frangi_a.std() < a.std()
+    assert hessian_a.std() > a.std()
+
+
+def test_values_decreased():
+    a = np.multiply(np.ones((3, 3)), 10)
+    frangi_a = frangi_filter(a)
+    hessian_a = hessian_filter(a)
+    assert_equal(frangi_a, np.zeros((3, 3)))
+    assert_equal(hessian_a, np.ones((3, 3)))
```

Comment:
I would also suggest a stricter test: crop out a small section (e.g. 40 x 40) of data.camera(), compute both filters, check result manually for sanity, and then save result alongside these tests for reference. Then call np.testing.assert_allclose() with some suitable tolerance for obtained results and reference.